test/llvm/vectorization/parallel_loop.chpl

```diff
@@ -0,0 +1,15 @@
+
+proc loop (A, B) {
+  // This loop should get vectorized when passed --vectorize
+  // CHECK: <4 x i32>
+  for i in vectorizeOnly(0..511) {
+      A[A[i]] = B[i];
+      A[i] = B[i+1];
+  }
```

Comment:
Adding loop vectorization (`llvm.loop.vectorize.enable`) is not necessary to get anything vectorized, it only makes sense when we want to **disable** vectorization of particular loop. What is important is adding `llvm.mem.parallel_loop_access`, because it greatly improves vectorization (we can remove runtime checks entirely and stuff gets vectorized anyway). I couldn't figure out working case by myself so I took this from llvm's test suite, `llvm.mem.parallel_loop_access` is necessary to get this vectorized, because it's very tricky and not obvious that there is no loop carried dependency.